src/turbomind/kernels/sampling_topk_kernels.cu

```diff
@@ -312,7 +312,7 @@ __global__ void topk_stage2_sampling(const int* __restrict topk_tmp_id_buf,
 }
 
 #define CASE_K(K_MIN, K_MAX, BLOCK_SIZE_1_, BLOCK_SIZE_2_, BLOCKS_PER_BEAM_)                                           \
-    case K_MIN ... K_MAX:                                                                                              \
```

Comment:
Will it slow the kernel down if change `switch-case` to `if`?

Reply:
msvc doesn't support range switch-case

Reply:
Can we make two versions and use OS macro to tell them apart?